// исходный текст программы сохранен в кодировке UTF-8 с сигнатурой

// xsyntax.cpp
// неработающий скелет программы для демонстрации синтаксиса исключений

class AClass                        // некий класс AClass
{
public:
	class AnError                   // класс, представляющий собой исключение
		{ };
	void Func()                     // метод класса AClass
	{
		if ( /* условие ошибки */ )
			throw AnError();        // генерируем исключение
	}
};

int main()                          // приложение
{
	try                             // блок попытки (блок try)
	{
		AClass obj1;                // взаимодействие с объектом класса AClass
		obj1.Func();                // тут может возникнуть ошибка
	}
	catch (AClass::AnError)         // обработчик исключения
	{                               // или «улавливающий блок» (блок catch)
		// сообщаем пользователю об ошибке и т.д.
	}
	
	return 0;
}